probe/plugins/reporter.go

```diff
@@ -0,0 +1,25 @@
+package plugins
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/weaveworks/scope/probe"
+	"github.com/weaveworks/scope/report"
+)
+
+// Reporter implements the Reporter interface for a plugin registry.
+func Reporter(pluginRegistry *Registry) probe.Reporter {
+	return probe.ReporterFunc("plugins", func() (report.Report, error) {
+		rpt := report.MakeReport()
+		pluginRegistry.Implementers("reporter", func(plugin *Plugin) {
+			pluginReport, err := plugin.Report()
+			if err != nil {
+				log.Errorf("plugins: error getting report from %s: %v", plugin.ID, err)
+				return
+			}
+			pluginReport.Plugins = pluginReport.Plugins.Add(plugin.PluginSpec)
```

Comment:
You probably want to add the plugin spec even if the plugin fails to report. Otherwise things may get more difficult to debug since, AFAIU, the plugin won't appear when querying the app's `/api` if it doesn't report anything in the last reporting window.

We may want to consider de-registering the plugin after certain amount of failures but I think that, while the plugin is registered, it should appear in `/api`. 

For debugging purposes, it may be worth adding a status field which can be be retrieved from `/api`.